### PR TITLE
Add pyright to pre-commit

### PR DESCRIPTION
**Pull request by rodrigodelazcano**
# Description
Added pyright to pre-commit and fixed errors such as : 

- Unbounded variables due to compound if statements.
- Solved `None` variable initialization with type annotation.
- Set the `agent_pos` attribute type to `numpy.array` for consistency. Environments with a fixed initial `agent_pos` assigned a `tuple` and when stepping this was converted to `numpy.array`.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
Sorry, I forgot to push my review

File: gym_minigrid/wrappers.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def reset(self, **kwargs):
         return self.env.reset(**kwargs)
 
 
-class ImgObsWrapper(gym.ObservationWrapper):
+class ImgObsWrapper(gym.core.ObservationWrapper):
```
[the remark this answers is not in the document]
Comment (rodrigodelazcano) [reply gym_minigrid/wrappers.py:105]: `ObservationWrapper` is not public in the gym package https://github.com/openai/gym/blob/dce35dfa92a5ede07aa7c5921733a1bf916e0100/gym/__init__.py#L22. `pyright` will through an error.

Should we modify `__all__` in the gym repo?

[the remark this answers is not in the document]
Comment (pseudo-rnd-thoughts) [reply gym_minigrid/wrappers.py:105]: That is surprising, I thought that would be ok. ignore me

File: gym_minigrid/envs/crossing.py [hunk 2]
```diff
@@ -30,7 +32,7 @@ def _gen_grid(self, width, height):
         self.grid.wall_rect(0, 0, width, height)
 
         # Place the agent in the top-left corner
-        self.agent_pos = (1, 1)
+        self.agent_pos = np.array((1, 1))
```
Comment (pseudo-rnd-thoughts) [inline gym_minigrid/envs/crossing.py:35]: This seems like an important type change from the old. Is there a reason for this?

Comment (rodrigodelazcano) [reply gym_minigrid/envs/crossing.py:35]: When given an initial `agent_pos` the type is `tuple`, but after the first step `agent_pos` is of type `numpy`. We should be consistent with the type.

**Comment by pseudo-rnd-thoughts** [comment]
Lgtm

**Pull request merged**